core/os/input_event.cpp

```diff
@@ -62,10 +62,16 @@ bool InputEvent::is_action(const StringName &p_action) const {
 
 bool InputEvent::is_action_pressed(const StringName &p_action) const {
 
-	return false; // InputMap::get_singleton()->event_is_action(Ref<InputEvent>(this),p_action);
+	if (is_action(p_action)) {
+		return is_pressed();
+	}
+	return false;
 }
 bool InputEvent::is_action_released(const StringName &p_action) const {
 
+	if (is_action(p_action)) {
+		return !is_pressed();
```

Comment:
Sorry if I'm being clueless on this matter but.... Why is !is_pressed() used for action_released? 
Shouldn't release be when pressed was true and then pressed becomes false? Otherwise as long as I'm not pressing the key, action_released() will always return true... which seems wrong